### PR TITLE
Streamline GetBlockSubsidy / fix coin cap comment

### PR DESCRIPTION
**Pull request by UdjinM6**
GetBlockSubsidy() can be streamlined a bit. Also fixing a comment about projected coin cap.

### CODE REVIEW

**Comment by UdjinM6** [comment]
This is going to break after #951, do not merge this yet, I'll fix it after #951 (which is more important)

**Comment by schinzelh** [comment]
Like expected, needs a rebase

**Comment by UdjinM6** [comment]
rebased, pls review :)

**Comment by tgflynn** [comment]
To me properly reviewing this change means making sure that these code changes never affect the actual value returned by the function (I assume no actual change to the computation is intended).  Unfortunately I don't have time for that right now since I have new sentinel functionality to implement.

I wonder if it's a good idea to be making a lot of refactoring changes now.  It seems to me there are still quite a few substantive questions about 12.1 that we need to figure out, maybe our time would be better spent on those questions at the moment.  

I think some projects try to reserve time periods for refactoring/improving code vs. making core changes.  Maybe we could consider putting some of these changes off to say early in the 12.2 cycle ?

**Comment by UdjinM6** [comment]
I would agree but I have most of refactoring postponed for like several month now waiting for the governance PRs and would like to clean that up, not redoing this again in 12.2. If you have no time to review each and every PR - that's fine, if no one have time - that's fine too, github doesn't charge fees for having PRs open :D

File: src/main.cpp [hunk 1]
```diff
@@ -1741,36 +1740,45 @@ NOTE:   unlike bitcoin we are using PREVIOUS block height here,
 */
 CAmount GetBlockSubsidy(int nPrevBits, int nPrevHeight, const Consensus::Params& consensusParams, bool fSuperblockPartOnly)
 {
-    double dDiff = (double)0x0000ffff / (double)(nPrevBits & 0x00ffffff);
-
-    /* fixed bug caused diff to not be correctly calculated */
-    if(nPrevHeight > 4500 || Params().NetworkIDString() != CBaseChainParams::MAIN) dDiff = ConvertBitsToDouble(nPrevBits);
```
Comment (crowning-) [inline src/main.cpp]: Why did you change this to mainnet?

Comment (UdjinM6) [inline src/main.cpp]: I didn't :)

Comment (schinzelh) [inline src/main.cpp]: Boolean logic: 

( A > 0 || B != C )

equals

(  !(A <= 0) || B != C)

equals

(  (A <= 0) && B = C)

:+1:

**Comment by schinzelh** [comment]
utACK 2339409

**Comment by crowning-** [comment]
Oops, somehow missed the &&  :+1: 
But I at least looked :-)
